xarray/core/dataarray.py

```diff
@@ -840,6 +840,99 @@ def swap_dims(self, dims_dict):
         ds = self._to_temp_dataset().swap_dims(dims_dict)
         return self._from_temp_dataset(ds)
 
+    def expand_dims(self, dim=None, axis=None):
```

Comment:
Consider implementing this on `xarray.Dataset` instead (looping over all data variables), using `_to_temp_dataset()`/`_from_temp_dataset()` to write the `DataArray` method: see `swap_dims` above for an example.